test/e2e/scripts/e2e.sh

```diff
@@ -43,25 +43,43 @@ fi
 
 oras_container_name="oras-e2e"
 upstream_container_name="oras-e2e-fallback"
+e2e_root="${repo_root}/test/e2e"
 echo " === preparing oras distribution === "
 run_registry \
-  ${repo_root}/test/e2e/testdata/distribution/mount \
+  ${e2e_root}/testdata/distribution/mount \
   ghcr.io/oras-project/registry:v1.0.0-rc.4 \
   $oras_container_name \
   $ORAS_REGISTRY_PORT
 
 echo " === preparing upstream distribution === "
 run_registry \
-  ${repo_root}/test/e2e/testdata/distribution/mount_fallback \
+  ${e2e_root}/testdata/distribution/mount_fallback \
   registry:2.8.1 \
   $upstream_container_name \
   $ORAS_REGISTRY_FALLBACK_PORT
 
+echo " === setup coverage instrumenting == "
+if [[ $GITHUB_REF_NAME == v* && $GITHUB_REF_TYPE == tag ]]; then
+    unset COVERAGE_DUMP_ROOT
+fi
+
+if ! [ -z ${COVERAGE_DUMP_ROOT} ]; then
+  rm ${e2e_root}/${COVERAGE_DUMP_ROOT} -rf
+fi
+
 echo " === run tests === "
-if ! ginkgo -r -p --succinct suite; then 
+ginkgo -r -p --succinct suite || fail=true
+
+if ! [ -z ${COVERAGE_DUMP_ROOT} ]; then
+  echo " === generating code cov report === "
+  go tool covdata textfmt -i="${e2e_root}/${COVERAGE_DUMP_ROOT}" -o ${e2e_root}/coverage.txt || true
+fi
```

Comment:
Should there be a make target and .gitignore entry for this?

Reply:
Changed as suggested.